### PR TITLE
Adds 3 scalars for use with NekRS simulations

### PR DESCRIPTION
**Pull request by ahuxford**
Mainly for use with NekRS RANS k-tau simulations where k and tau are scalar01 and scalar02. I added an additional scalar03 in case an additional passive scalar transport is of interest.

### CODE REVIEW

**Comment by aprilnovak** [comment]
This looks good! There's a few features we should add to be sure the documentation is up to date and one sanity check:

- [ ] An error test that throws the correct error if the user puts `scalar03`, but doesn't have 4 scalars - otherwise, this option would seg fault
- [ ] In the input parameters for some of the classes, we should add the new options to the documentation string, i.e. 

```
params.addRequiredParam<MooseEnum>("field",
                                     getNekFieldEnum(),
                                     "Field to integrate; "
                                     "options: velocity_x, velocity_y, velocity_z, "
                                     "velocity_component, velocity, temperature, pressure, unity");
```

- [ ] Add the new options to `field_specs.md` so that they show up in the documentation (also add the caveat that we're not dimensionalizing these fields, in `nondimensional.md`

**Comment by ahuxford** [comment]
Yes good points. I'll update this PR with the changes so we add everything at once for these added scalars.

**Comment by ahuxford** [comment]
I think I addressed all of your feedback.

For the tests, I used `test/tests/nek_errors/invalid_field` as a template to test Nek postProcessors and Nek UserObjects.

I did not add the scalars to the `NekRSStandaloneProblem` options for `outputs` yet, so there's no test for that (unlike the test for temperature). 

I do plan to add the scalars to the `outputs` options in a future PR though, because that will be very useful.

File: src/postprocessors/NekPostprocessor.C [hunk 1]
```diff
@@ -54,6 +54,18 @@ NekPostprocessor::checkValidField(const field::NekFieldEnum & field) const
   if (!nekrs::hasTemperatureVariable() && field == field::temperature)
     mooseError("This postprocessor cannot set 'field = temperature' "
                "because your Nek case files do not have a temperature variable!");
+
```
[the remark this answers is not in the document]
Comment (ahuxford) [reply src/postprocessors/NekPostprocessor.C:57]: Yes that'd be better. I'll move the errors to solutionPointer. I'll move the temperature over to there also right now.

**Comment by ahuxford** [comment]
I think this should be ready once the tests finish.

**Comment by aprilnovak** [comment]
Thanks for this contribution!

**Pull request merged**